Add theming for Unity Greeter

HEXcube opened the pull request:
Arc (and most GTK themes not bundled with Ubuntu) leaves **Unity Greeter** (Ubuntu's default LightDM greeter) unthemed. :worried: Like this:

![ArcThemedUnityGreeter](https://cloud.githubusercontent.com/assets/3755396/14580874/b1d37178-03fa-11e6-8183-f396cee0532b.png)

*budgie-remix* is switching to Unity Greeter and wanted to ensure it looks good on Arc theme. So, I just copied the contents of `unity-greeter.css` from [Ubuntu Themes upstream repo](https://bazaar.launchpad.net/~ubuntu-art-pkg/ubuntu-themes/trunk/view/head:/Ambiance/gtk-3.0/apps/unity-greeter.css), converted to SASS using [css2sass](https://css2sass.herokuapp.com) and added it to the existing `_unity.scss` in Arc. :relaxed: I added the same content to all 4 versions (GTK3.20 to GTK3.14), coz as u can see in [unity-greeter's file history](https://bazaar.launchpad.net/~ubuntu-art-pkg/ubuntu-themes/trunk/changes?filter_file_id=unitygreeter.css-20120206150958-niclp6mwf7mblqej-1), it wasn't changed since Ubuntu 15.04 (which used GTK3.14). With the addition of Unity Greeter CSS code (and changing background via gsettings) Unity Greeter on Arc looks like this:

![ArcUnthemedUnityGreeter](https://cloud.githubusercontent.com/assets/3755396/14580875/c60ff274-03fa-11e6-934a-e03c72b89994.png)

horst3180 commented:
Thank you for adding this :)

There a some points I'd like to see fixed, though:

* You didn't regenerate the css files from the scss sources. See [here](https://github.com/horst3180/arc-theme/blob/master/HACKING.md) for more information.

* In Sass `alpha()` is used to get the alpha value of a color, not to set it. Use `transparentize()` instead.

* `shade()` won't work either. Use `lighten()` and `darken()` instead.

* Please don't use this css2sass converter, it generates ugly code. Just paste the css as it is into the scss file. You don't have to nest everything, although a bit of reorganization/nesting would be nice.

HEXcube commented:
@horst3180  Thanks for reviewing my commit! :relaxed: I'll make the necessary changes on the scss and generate css files. And this time, I'll do it manually instead of using that online css to sass converter, though I'm bit of a newbie in css/sass field. :relieved:

HEXcube commented:
K.. I added a new commit with css generated from sass, alpha() and shade() methods replaced. :relaxed:  As my scss and css skill are at n00b level, I didn't attempt to manually convert the [original one from Ambiance theme](https://bazaar.launchpad.net/~ubuntu-art-pkg/ubuntu-themes/trunk/view/head:/Ambiance/gtk-3.0/apps/unity-greeter.css).

And I don't know why it's showing 3 commits instead of two. :open_mouth: I just tried to merge all upstream commits before pushing, according to this guide: https://help.github.com/articles/syncing-a-fork/

vooze commented:
I have tried both versions you made, and looking at your screenshot your icons are a bit weird. Also the clocks color is hard to read now.

HEXcube commented:
@vooze  Yeah, the panel looks wierd. :worried: Do u have any idea how to fix it? Can u have a look at https://github.com/HEXcube/arc-theme/blob/master/common/gtk-3.0/3.18/sass/_unity.scss#L61 and gimme some suggestions? :confused:

vooze commented:
@HEXcube I don't unfortunately.

HEXcube commented:
@vooze It's alright! I'll figure it out, eventually :smile:

horst3180 commented:
Just a shot in the dark. Try `.lightdm.menubar { color: white; }` for the clock.

HEXcube commented:
@horst3180 Thanx! I'll try that. I'm currently rewriting Sass manually from the original CSS like u suggested. :relaxed:

HEXcube commented:
@horst3180 K.. removed excessive nesting and rebased css from [upstream](https://bazaar.launchpad.net/~ubuntu-art-pkg/ubuntu-themes/trunk/view/head:/Ambiance/gtk-3.0/apps/unity-greeter.css). :relaxed:

vooze commented:
Did you fix the icons? :-)

HEXcube commented:
@vooze Haven't figured that out yet! :worried: But I followed @horst3180's suggestion and fixed clock text color! :smiley: See:

![afterthemingunitygreeter](https://cloud.githubusercontent.com/assets/3755396/14690791/8223b55e-076b-11e6-9206-90a2d7637215.png)

vooze commented:
Great. I looked at the icons my self.. Its weird because it's really the same code as Ambiance unity greeter.

HEXcube commented:
@vooze Yeah, that's what bugs me too! :confused: Since it works fine with Ambiance and Radiance, it's supposed to work fine on other GTK themes too.

I tried to find some documentation for the stuff inside `unity-greeter.css` but found this instead: http://askubuntu.com/questions/692644/documentation-for-theming-of-unity-greeter/696147#696147 . It suggests inspecting the vala source files of [unity-greeter](http://bazaar.launchpad.net/~unity-greeter-team/unity-greeter/trunk/files) to find class declarations. I'm gonna give it a try. :relaxed:

HEXcube commented:
Good news! Finally fixed panel icons. Whew! :relieved:

On **Ubuntu 16.04LTS**:

![ArcUnityGreeterTheme](https://cloud.githubusercontent.com/assets/3755396/14763347/90269710-09b0-11e6-8fde-c479724d5112.png)

On **budgie-remix**:

![budgie-remixGreeter](https://cloud.githubusercontent.com/assets/3755396/14763340/4580804a-09b0-11e6-8c56-0b71065a8b60.png)

So @horst3180 , is it merge ready now? :relaxed:

vooze commented:
@HEXcube I still can't get it to work..

I have tried both changing it in gtk3-0.css and also a unity-greeter.css and import that.

How are you changing it?
sudo nano /usr/share/glib-2.0/schemas/com.canonical.unity-greeter.gschema.xml - edit it to arc-theme-something
sudo glib-compile-schemas /usr/share/glib-2.0/schemas/

Or am I missing something?

HEXcube commented:
@vooze Are u on Ubuntu? If so, create an override file in `/usr/share/glib-2.0/schemas/10_unity-greeter.gschema.override` and add this in it: 

```
[com.canonical.unity-greeter]
theme-name='Arc-Dark'
background-color='#99004C'
```

If you're on budgie-remix, get the latest budgie-remix-lightdm-theme package from [our PPA](https://launchpad.net/~budgie-remix/+archive/ubuntu/ppa), install it and add this to the content in `/etc/lightdm/lightdm.conf.d/50-budgie-desktop.conf`:

```
greeter-session=unity-greeter
```

vooze commented:
@HEXcube Still not working.. Really weird. I get the same as your first image. White login-field and weird icons.. I can see that it's changing the theme from Ambiance, but it's not really applying the lightdm-theme I think.

Can you see something I did wrong in my gtk3-0.css? https://gist.github.com/vooze/cd3b1b26a98ffc7dfc46d79f50982f1d

I really can't understand why It's just not working :)

HEXcube commented:
@vooze Can  u show me a screenshot? :confused:

vooze commented:
@HEXcube Can't really take a screenshot (or at least I don't know how from lightdm)

here is from my phone-cam: https://goo.gl/photos/LAJTwwW4HBYWqry5A

vooze commented:
@HEXcube nevermind! it works now.. I'm an idiot.. Don't even ask what I did wrong ;)
Thank you so much!

HEXcube commented:
Glad to hear that! :relieved: I was worried I still have something left to fix :laughing:

vooze commented:
@HEXcube Last thing.. I have my own branch of arc-theme for ubuntu with some unity-fixes. Can I use your screenshot and credit you for the lightdm theme? https://github.com/vooze/arc-darker-ubuntu

HEXcube commented:
@vooze Yeah, sure! :relaxed: And it's not *LightDM* theme, but *Unity Greeter* theme to be exact! :stuck_out_tongue_winking_eye:

And your Arc adapted for Unity seems interesting! :+1:

vooze commented:
@HEXcube Awesome. I have credited you here: https://github.com/vooze/arc-darker-ubuntu/blob/master/README.md

Thank you so much again! :smile:

vooze commented:
@HEXcube Did not see your last message. But yes, I'm trying to start with fixing the unity topbar, but there are so many files to edit, and just joined github, Need to figure out how.

horst3180 commented:
@HEXcube Looks good to me. I think it's ready to merge now.

HEXcube commented:
@horst3180  Thanx! :kissing_heart: budgie-remix team'll contribute back if we do more improvements. :relaxed:

leoheck commented:
I think the color of the hostname font (top-left) needs to be updated too.

cmcmcmcm123 commented:
@HEXcube I was thinking of overriding the icon theme as well, but I have no clue to do so. Should I just do it the same way you overrided the theme?

[com.canonical.unity-greeter]
icon-theme-name='Paper'

also, should I just append this on the recently, override file made?
